### ISSUE TITLE
Senseless comparison

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#966]
Look this example:

```php
function example(User $user) {
    return $user > 0;
}
```

It is a senseless comparison once that `$user` is needly `User`, that is an object, and will never works like that. So in cases like that, maybe we can notify user about that.

* If `$var` is an object:
    * Disallow any comparison (`>`, `>=`, `<`, `<=`, `<=>`);
    * Disallow null comparison (`=== null`, `!== null`, `== null`, `!= null`);
        * Except if it is a nullable object;

For the case where object is nullable as in the example below:

```php
function example(?User $user) {
    return $user !== null;
}
```

I think that it could suggest to replace it with a more simplified boolean:

```
return $user === null
    to return (bool) $user;

return $user !== null
    to return !$user;

if ($user === null)
    to if ($user)

if ($user !== null)
    to if (!$user)
```

(Maybe this last should be a separated issue, tell if you prefer that.)

**Comment by kalessil** [comment]
Unfortunately, PhpStorm types resolver is not yet ready for supporting this inspection.
Fixing reported resolver bugs are geting postponed, so I cannot deliver anything as good as described.
